apps/server/src/infra/sync/tsp/tsp-sync.strategy.ts

```diff
@@ -122,4 +144,75 @@ export class TspSyncStrategy extends SyncStrategy {
 
 		this.logger.info(new TspSyncedUsersLoggable(results.length));
 	}
+
+	private async migrateTspTeachers(system: System): Promise<{ total: number }> {
+		const tspTeacherIds = await this.tspSyncService.fetchTspTeacherMigrations(system);
+		this.logger.info(new TspTeachersFetchedLoggable(tspTeacherIds.length));
+
+		const teacherMigrationPromises = tspTeacherIds.map(({ lehrerUidAlt, lehrerUidNeu }) =>
+			this.migrationLimit(async () => {
+				if (lehrerUidAlt && lehrerUidNeu) {
+					await this.migrateTspUser(lehrerUidAlt, lehrerUidNeu, system.id);
+					return true;
+				}
+				return false;
+			})
+		);
+
+		const migratedTspTeachers = await Promise.allSettled(teacherMigrationPromises);
+
+		const total = migratedTspTeachers.filter((result) => result.status === 'fulfilled' && result.value === true).length;
+		this.logger.info(new TspTeachersMigratedLoggable(total));
+
+		return { total };
+	}
+
+	private async migrateTspStudents(system: System): Promise<{ total: number }> {
+		const tspStudentIds = await this.tspSyncService.fetchTspStudentMigrations(system);
+		this.logger.info(new TspStudentsFetchedLoggable(tspStudentIds.length));
+
+		const studentMigrationPromises = tspStudentIds.map(({ schuelerUidAlt, schuelerUidNeu }) =>
+			this.migrationLimit(async () => {
+				if (schuelerUidAlt && schuelerUidNeu) {
+					await this.migrateTspUser(schuelerUidAlt, schuelerUidNeu, system.id);
+					return true;
+				}
+				return false;
+			})
+		);
+
+		const migratedStudents = await Promise.allSettled(studentMigrationPromises);
+
+		const total = migratedStudents.filter((result) => result.status === 'fulfilled' && result.value === true).length;
+		this.logger.info(new TspStudentsMigratedLoggable(total));
+
+		return { total };
+	}
```

Comment:
The only difference between these methods is the call to get the ids. Can we combine these methods?

Reply:
We have discussed this and 2 methods make more sense.